### PR TITLE
Allows configuring the profile used during the native tests

### PR DESCRIPTION
**Pull request by cescoffier**
Related to https://github.com/quarkusio/quarkus/issues/4371

### CODE REVIEW

File: docs/src/main/asciidoc/building-native-image-guide.adoc [hunk 1]
```diff
@@ -209,6 +209,13 @@ duration can be changed using the `quarkus.test.native-image-wait-time` system p
 to 300 seconds, use: `./mvnw verify -Pnative -Dquarkus.test.native-image-wait-time=300`.
 ====
 
+By default, native tests runs using the `prod` profile.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/building-native-image-guide.adoc]: IMO, that's confusing. Especially taking into account the way we are writing the tests for the quickstarts where the native test simply extends the JVM one. So if you have something in the test profile, the JVM one will pass and the native one will fail (unless you explicitly enable the profile for it, of course).

File: test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java [hunk 2]
```diff
@@ -35,10 +36,12 @@ public NativeImageLauncher(Class<?> testClass) {
         this(testClass,
                 ConfigProvider.getConfig().getOptionalValue("quarkus.http.test-port", Integer.class).orElse(DEFAULT_PORT),
                 ConfigProvider.getConfig().getOptionalValue("quarkus.test.native-image-wait-time", Long.class)
-                        .orElse(DEFAULT_IMAGE_WAIT_TIME));
+                        .orElse(DEFAULT_IMAGE_WAIT_TIME),
+                ConfigProvider.getConfig().getOptionalValue("quarkus.test.native-image-profile", String.class)
```
Comment (gsmet) [inline test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java]: Please add it here: https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java#L74 .

File: test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java [hunk 3]
```diff
@@ -60,6 +64,9 @@ public void start() throws IOException {
         args.add("-Dquarkus.http.port=" + port);
         args.add("-Dtest.url=" + TestHTTPResourceManager.getUri());
         args.add("-Dquarkus.log.file.path=" + PropertyTestUtil.getLogFileLocation());
+        if (profile != null) {
+            args.add("-Dquarkus.profile=" + profile);
```
Comment (gsmet) [inline test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java]: `prod` will be the default value here? I'm not totally sure.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java (hunk 3 above)
Comment (cescoffier) [reply test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java]: Yes, but there is an ongoing discussion about what should be the default.
(https://github.com/quarkusio/quarkus/issues/4371)

So, I've decided to not override the current default or next-to-be default. It's something we could revisit. 

So, this PR just offers the user the possibility to customize it.

File: docs/src/main/asciidoc/building-native-image-guide.adoc (hunk 1 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/building-native-image-guide.adoc]: @aloubyansky I learned that the hard way (dropping my prod database, oupsy...)
More discussion is happening here: https://github.com/quarkusio/quarkus/issues/4371

File: test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java (hunk 2 above)
Comment (cescoffier) [reply test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java]: Fixed. I didn't know about this actually.

**Review by aloubyansky — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

File: docs/src/main/asciidoc/building-native-image-guide.adoc [hunk 4]
```diff
@@ -209,6 +209,13 @@ duration can be changed using the `quarkus.test.native-image-wait-time` system p
...
+This can be overridden using the `quarkus.test.native-image-profile` property.
+For example, in your `application.properties` file, add: `quarkus.test.native-image-profile=test`.
+Alternatively, you can run your tests with: `./mvnw verify -Pnative -Dquarkus.test.native-image-profile=test`.
+However, don't forget that when the native executable is built the `prod` profile is enabled.
```
Comment (emmanuelbernard) [inline docs/src/main/asciidoc/building-native-image-guide.adoc]: I find this sentence confusing. Are you saying that your tests will run with the test profile but pointing to the image that was built as prod.
Or that your image will be build with test and that should be compatible with the "prod" environment you have in mind.

My fear is that this is really inside baseball here and that's typically this is written for the implementer to feel happy that he has covered all cases than in the spirit of helping the user. If we see different use cases, we should enunciate them better so people could connect to the problem + solution.

This is about improvement so if we cannot make it I won't die for it.

**Pull request merged**